modules/carto/src/layers/heatmap.ts

```diff
@@ -0,0 +1,181 @@
+import type {ShaderPass} from '@luma.gl/shadertools';
+import {random} from '@luma.gl/shadertools';
+import {colorCategories} from '@deck.gl/carto';
+const glsl = (s: TemplateStringsArray) => `${s}`;
+
+/**
+ * @filter       Heatmap
+ * @param radius Blur radius, controls smoothness of heatmap
+ * @param rangeScale Scale factor to apply to values prior to applying color scale
+ * @param color1-6 Colors to use in color scale
+ */
+
+const fs = glsl`\
+uniform heatmapUniforms {
+  float radius;
+  float rangeScale;
+  vec3 color1;
+  vec3 color2;
+  vec3 color3;
+  vec3 color4;
+  vec3 color5;
+  vec3 color6;
+} heatmap;
+
+// Controls quality of heatmap, larger values increase quality at expense of performance
+const float SUPPORT = 8.0;
```

Comment:
Totally fine to leave this out of the PR for now, just a thought — I suspect we may want to parameterize this at some point. 256 samples per fragment feels like a lot on mobile (maybe not though? I haven't tested on mobile yet), and (to my eye) the heatmap looks very good with ~64 samples too.

Reply:
I think it should be possible to separate the 2D operation into two passes also, so instead of `O(n*n)` it'll be `O(2n)` - so 32 samples rather than 256

The nice thing about the mobile use case is that the smaller the screen, the cheaper the post-process effect is

Reply:
Managed to do this 😊 https://github.com/visgl/deck.gl/pull/8703/commits/0c605d7bde623e17d17206bcdc18756f6ee77593

Reply:
Really nice! On my computer the performance looks excellent. Testing on my phone (iPhone 11) there is some interaction delay, but it appears to be more caused by tesselation in `solid-polygon-layer.ts`, not anything in your heatmap shader – so that seems like a good sign that the shader is optimized enough! 🚀

Reply:
Yes, I tried it against using the `RasterLayer` and the performance is indeed better